tekton/pipeline_run.go

```diff
@@ -135,6 +138,34 @@ func (r *ReleasePipelineRun) WithEnterpriseContractPolicy(enterpriseContractPoli
 	return r
 }
 
+// copyPipelinesAsCodeMeta copies key/value pairs in src adding them to dst.
+// When a key in src is already present in dst, no copy happens.
+// When the src key does not contain the prefix string, no copy happens.
+func copyPipelinesAsCodeMeta(src, dst map[string]string) {
+	for key, val := range src {
+		if _, err := dst[key]; !err && strings.HasPrefix(key, pipelinesAsCodeMetaPrefix) {
+			dst[key] = val
+		}
+	}
+}
+
+// WithPipelinesAsCodeMeta conditionally copies Labels and Annotations from Release to PipelineRun
+// When Annotations or Labels maps do not exist, then they are created.
+// Only things who match pipelinesAsCodeMetaPrefix are allowed to be copied.
+func (pipelineRun *ReleasePipelineRun) WithPipelinesAsCodeMeta(release *v1alpha1.Release) *ReleasePipelineRun {
+	if pipelineRun.ObjectMeta.Annotations == nil {
+		pipelineRun.ObjectMeta.Annotations = make(map[string]string)
+	}
+	copyPipelinesAsCodeMeta(release.ObjectMeta.Annotations, pipelineRun.ObjectMeta.Annotations)
+
+	if pipelineRun.ObjectMeta.Labels == nil {
+		pipelineRun.ObjectMeta.Labels = make(map[string]string)
```

Comment:
This makes the With functions order dependant. If WithReleaseAndApplicationLabels get called  after this function, labels will get removed.

Reply:
That makes sense. I did try to make it safe, not clobbering things already setup. Regardless, I've moved the new code into `WithReleaseAndApplicationLabels` and per another remark renamed that to `WithReleaseAndApplicationMetadata`. PTAL